### PR TITLE
feat: screenshots aren't secure modal

### PR DESCRIPTION
**Pull request by ice-ajax**
## Additional Notes
- Tested on Android 13 (no dedicated screenshot detection API), Android 14 and iOS
- Current limitation: screenshot detection doesn't work for Android version prior to 14 if external storage permission is not granted. App asks for it the first time user tries to upload any media. After it is granted, everything works as expected.

## Type of Change
- [ ] Bug fix
- [x] New feature
- [ ] Breaking change
- [ ] Refactoring
- [ ] Documentation
- [ ] Chore

## Videos (if applicable)

Android:
<video src="https://github.com/user-attachments/assets/16fd49dc-a947-4ed8-81f9-ba1c784b117c"/>

iOS:
<video src="https://github.com/user-attachments/assets/52ffdd32-38d9-4847-8e45-63782cb85f01"/>

### CODE REVIEW

File: pubspec_overrides.yaml [hunk 1]
```diff
@@ -1,5 +1,7 @@
-# melos_managed_dependency_overrides: ion_identity_client,js
+# melos_managed_dependency_overrides: ion_identity_client,js,ion_screenshot_detector
 dependency_overrides:
   ion_identity_client:
     path: packages/ion_identity_client
+  ion_screenshot_detector:
```
[the remark this answers is not in the document]
Comment (ice-ajax) [reply pubspec_overrides.yaml:5]: It is auto generated by melos when running `melos bootstrap`

File: packages/ion_screenshot_detector/test/simple_test.dart [hunk 2]
```diff
@@ -0,0 +1,12 @@
+// SPDX-License-Identifier: ice License 1.0
+
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
```
[the remark this answers is not in the document]
Comment (ice-ajax) [reply packages/ion_screenshot_detector/test/simple_test.dart:5]: Yes, at least a single test case inside a package is needed for our `melos run test` script to pass.

File: lib/app/features/protect_account/backup/views/pages/create_recover_key_page/hooks/use_on_screenshot.dart [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
+import 'package:ion_screenshot_detector/ion_screenshot_detector.dart';
+
+void useOnScreenshot(VoidCallback callback) {
+  useEffect(
+    () {
+      final detector = IonScreenshotDetector()
+        ..startListening((_) {
```
[the remark this answers is not in the document]
Comment (ice-ajax) [reply lib/app/features/protect_account/backup/views/pages/create_recover_key_page/hooks/use_on_screenshot.dart:11]: stream subscription is created inside the `IonScreenshotDetector` and get disposed of when the dispose function of a detector is called inside a hook's dispose function (`detector.disposed()`);

File: lib/app/features/protect_account/backup/views/pages/create_recover_key_page/components/create_recovery_key_success_state.dart [hunk 4]
```diff
@@ -20,6 +24,13 @@ class CreateRecoveryKeySuccessState extends StatelessWidget {
   Widget build(BuildContext context) {
     final locale = context.i18n;
 
+    useOnScreenshot(
+      () => showSimpleBottomSheet<void>(
```
[the remark this answers is not in the document]
Comment (ice-ajax) [reply lib/app/features/protect_account/backup/views/pages/create_recover_key_page/components/create_recovery_key_success_state.dart:28]: Done

**Review by ice-alcides — APPROVED** [review]

**Pull request merged**